File: modules/profile_store.js
export async function getStoredProfileBundle() {
  const stored = await chrome.storage.local.get(["profile", "resumeDetails", "ollamaModel", "reviewedJobs"]);
  return enrichProfileBundle({
    profile: stored.profile || {},
    resumeDetails: stored.resumeDetails || {},
    ollamaModel: stored.ollamaModel || "",
    reviewedJobs: stored.reviewedJobs || []
  });
}

export async function saveReviewedJobs(jobs) {
  await chrome.storage.local.set({ reviewedJobs: jobs || [] });
}

export function hasProfileData(bundle) {
  const profile = bundle?.profile || {};
  const details = bundle?.resumeDetails || {};
  return Boolean(
    Object.values(profile).some((value) => String(value || "").trim()) ||
    Object.values(details).some((value) => Array.isArray(value) && value.length)
  );
}

export const DEFAULT_TARGET_ROLES = [
  "Project Coordinator",
  "Assistant Project Manager",
  "Operations Coordinator",
  "Business Operations Analyst",
  "Risk Analyst",
  "Program Coordinator",
  "Implementation Coordinator",
  "Real Estate Operations"
];

export const DEFAULT_TARGET_LOCATIONS = ["Bay Area", "New York", "New Jersey", "Remote"];

export const DEFAULT_VISA_CONSTRAINTS = {
  status: "F-1 OPT",
  filterCitizenship: true,
  filterSecurityClearance: true,
  filterNoSponsorship: true
};

export function enrichProfileBundle(bundle) {
  const profile = bundle.profile || {};
  return {
    ...bundle,
    targetRoles: bundle.targetRoles?.length ? bundle.targetRoles : [
      profile.desiredTitle,
      profile.currentTitle,
      ...DEFAULT_TARGET_ROLES
    ].filter(Boolean),
    targetLocations: bundle.targetLocations?.length ? bundle.targetLocations : DEFAULT_TARGET_LOCATIONS,
    visaConstraints: bundle.visaConstraints || DEFAULT_VISA_CONSTRAINTS
  };
}

File: job_agent/profile_store.py
import json
from pathlib import Path

TARGET_ROLES = [
    "Project Coordinator",
    "Assistant Project Manager",
    "Operations Coordinator",
    "Business Operations Analyst",
    "Risk Analyst",
    "Program Coordinator",
    "Implementation Coordinator",
    "Real Estate Operations",
]

TARGET_LOCATIONS = ["Bay Area", "New York", "New Jersey", "Remote"]

VISA_CONSTRAINTS = {
    "status": "F-1 OPT",
    "filter_citizenship": True,
    "filter_security_clearance": True,
    "filter_no_sponsorship": True,
}


def load_profile(path="profile_store.json"):
    data = json.loads(Path(path).read_text(encoding="utf-8")) if Path(path).exists() else {}
    data.setdefault("profile", {})
    data.setdefault("resumeDetails", {})
    data.setdefault("targetRoles", TARGET_ROLES)
    data.setdefault("targetLocations", TARGET_LOCATIONS)
    data.setdefault("visaConstraints", VISA_CONSTRAINTS)
    return data


def profile_text(profile):
    details = profile.get("resumeDetails", {})
    pieces = profile.get("targetRoles", []) + profile.get("targetLocations", [])
    for value in profile.get("profile", {}).values():
      if isinstance(value, str):
        pieces.append(value)
    for group in ["skillsAndTools", "experienceEntries", "projectEntries", "educationEntries", "certificationEntries"]:
        for item in details.get(group, []) or []:
            pieces.extend(str(value) for value in item.values() if value)
    if profile.get("rawText"):
        pieces.append(profile["rawText"])
    return "\n".join(pieces)

